samples/apps/copilot-chat-app/webapp/src/components/utils/TextUtils.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Function to detect and convert URLs within a string into clickable links.
+ * It wraps each link matched with anchor tags and applies safe href attributes.
+ */
+export function convertToAnchorTags(htmlString: string) {
+    // Regular expression to match links, excluding any HTML tags at the end
+    // Since response from bot is plain text, sometimes line breaks and other html tags are included in the response for readability.
+    var linkRegex = /(?:https?):\/\/(\w+:?\w*)?(\S+)(:\d+)?(\/|\/([\w#!:.?+=&%!\-/]))?(?=(<br|<p|<div|<span)\s*\/>|$)/g;
```

Comment:
What if it is an http:// link?

Reply:
The ? after the s matches 0 or more characters
![image](https://github.com/microsoft/semantic-kernel/assets/125500434/40a77527-8c81-4ba1-a9ac-e3fe21a95524)